concent_api/signing_service_testing_common.py

```diff
@@ -0,0 +1,236 @@
+    if count_fails.get_fails() > 0:
+        count_fails.print_fails()
+
+    print("END")
+
+
+def check_if_required_components_are_running(components, config):
+    """ Main tests runner, prepares test config, gather test functions, executes them, show results. """
+    if Components.CONCENT_API in components:
+        if not is_concent_api_is_running(**dict(config.items(Components.CONCENT_API.value))):
+            raise ComponentConnectionError(
+                f'Unable to connect to ConcentAPI with config: {config.items(Components.CONCENT_API.value)}'
+            )
+    if Components.MIDDLEMAN in components:
+        if not is_middleman_is_running(**dict(config.items(Components.MIDDLEMAN.value))):
+            raise ComponentConnectionError(
+                f'Unable to connect to MiddleMan with config: {config.items(Components.MIDDLEMAN.value)}'
+            )
+    if Components.ETHEREUM_BLOCKCHAIN in components:
+        if not is_ethereum_blockchain_is_running(**dict(config.items(Components.ETHEREUM_BLOCKCHAIN.value))):
+            raise ComponentConnectionError(
+                f'Unable to connect to Ethereum blockchain with config: {config.items(Components.ETHEREUM_BLOCKCHAIN.value)}'
+            )
+
+
+def is_concent_api_is_running(api_url):
+    """ Verifies if Concent API server instance is running on given API url. """
+    try:
+        response = requests.post(
+            api_url,
+            data=Ping().serialize(),
+            headers={
+                'Content-Type': 'application/octet-stream',
+            },
+            verify=False,
+        )
+        return (
+            response.json()['error'] == 'Unknown message type' and
+            response.json()['error_code'] == 'message.unknown'
+        )
+    except requests.ConnectionError:
+        return False
+
+
+def is_middleman_is_running(host, port):
+    """ Verifies if MiddleMan server instance is running on given host and port. """
+    client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    client_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, True)
+
+    try:
+        client_socket.connect((host, int(port)))
+        return True
+    except ConnectionRefusedError:
+        return False
+    finally:
+        client_socket.close()
+
+
+def is_ethereum_blockchain_is_running(geth_address, concent_ethereum_address):
+    """ Verifies if Ethereum Blockchain is running and can be connected. """
+    sci_client = new_sci_rpc(
+        rpc=geth_address,
+        address=concent_ethereum_address,
+        chain=chains.RINKEBY,
+        storage=JsonTransactionsStorage(filepath=Path('./storage.json')),
+    )
+    transaction_count = sci_client.get_transaction_count()
+    return isinstance(transaction_count, int)
+
+
+def send_message_to_middleman_and_receive_response(message, config, signing_service_public_key, concent_private_key):
+    """ Sends message to MiddleMan using MiddleMan protocol and retrieves response. """
```

Comment:
I agree that they might look duplicated but here are my reasons:
1. It is better to put anything in comment that nothing.
2. It can be just a placeholder, which can be modified and extended later.

Generally I think it would be good to unify this. I also think we should make more comments in our code, not only for functions/methods but before every bigger block of code.

Reply:
:+1: agree